starport/pkg/tarball/tarball.go

```diff
@@ -0,0 +1,65 @@
+package tarball
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"io"
+	"path/filepath"
+)
+
+var (
+	// ErrGzipFileNotFound the file not found in the gzip
+	ErrGzipFileNotFound = errors.New("file not found in the gzip")
+	// ErrInvalidGzipFile the gzip file is invalid
+	ErrInvalidGzipFile = errors.New("invalid gzip file")
+)
+
+// IsTarball checks if the file is a tarball
+func IsTarball(tarball []byte) error {
```

Comment:
```suggestion
func IsTarball(tarball io.Reader) error {
```

We should work with io.Reader/io.Writer's because these files might get huge.